lib/Workflow.pm

```diff
@@ -427,86 +427,104 @@ This documentation describes version 2.05 of Workflow
-         <action name="upload file" resulting_state="uploaded" />
-     </state>
-     <state name="uploaded" autorun="yes">
-         <action name="verify file" resulting_state="verified file">
-              <!-- everyone other than 'CWINTERS' must verify -->
-              <condition test="$context->{user} ne 'CWINTERS'" />
-         </action>
-         <action name="null" resulting_state="annotated">
-              <condition test="$context->{user} eq 'CWINTERS'" />
-         </action>
-     </state>
-     <state name="verified file">
-         <action name="annotate">
-             <condition name="can_annotate" />
-         </action>
-         <action name="null">
-             <condition name="!can_annotate" />
-         </action>
-     </state>
-     <state name="annotated" autorun="yes" may_stop="yes">
-         <action name="null" resulting_state="finished">
-            <condition name="completed" />
-         </action>
-     </state>
-     <state name="finished" />
- </workflow>
 
  # Defines actions available to the workflow
- my $action_conf    = 'action.xml';
+ my $action_conf    = 'action.yaml';
+
+ # contents of 'action.yaml'
+
+ action:
+ - name: "upload file"
+   class: MyApp::Action::Upload
+   field:
+   - name: path
+     label: "File Path"
+     is_required: yes
+     description: |-
+       Path to file
+ - name: "verify file"
+   class: MyApp::Action::Verify
+   validator:
+   - name: filesize_cap
+     value: "$file_size"
+ - name: annotate
+   class: MyApp::Action::Annotate
+ - name: "null"
+   class: Workflow::Action::Null
+
+ # end of action.yaml
 
- # contents of 'action.xml'
 
- <actions>
-     <action name="upload file" class="MyApp::Action::Upload">
-         <field name="path" label="File Path"
-                description="Path to file" is_required="yes" />
-     </action>
+ # Defines conditions available to the workflow
+ my $condition_conf = 'condition.yaml';
 
-     <action name="verify file" class="MyApp::Action::Verify">
-         <validator name="filesize_cap">
-             <arg>$file_size</arg>
-         </validator>
-     </action>
+ # contents of 'condition.yaml'
 
-     <action name="annotate"    class="MyApp::Action::Annotate" />
+ condition:
+ - name: can_annotate
+   class: MyApp::Condition::CanAnnotate
 
-     <action name="null"        class="Workflow::Action::Null" />
- </actions>
+ # end of condition.yaml
 
- # Defines conditions available to the workflow
- my $condition_conf = 'condition.xml';
 
- # contents of 'condition.xml'
+ # Defines validators available to the actions
+ my $validator_conf = 'validator.yaml';
 
- <conditions>
-     <condition name="can_annotate"
-                class="MyApp::Condition::CanAnnotate" />
- </conditions>
+ # contents of 'validator.yaml'
 
- # Defines validators available to the actions
- my $validator_conf = 'validator.xml';
+ validator:
+ - name: filesize_cap
+   class: MyApp::Validator::FileSizeCap
+   param:
+   - name: max_size
```

Comment:
Would have been nice to have had a hash instead of an array. Are we too far on the 2.x release line?

```yaml
   param:
     max_size: 20M
```

Thinking about the options: maybe we can support *both* arrays (of hashes with the name/value keys) and plain hashes?

Reply:
Sound like a path if we do as you suggest, then we could phase out plain hashes with 3.0, but allow for both in 2.X - if I understand your comment correctly